### PR TITLE
feat(recipe-list): bare-bones recipe list

### PR DESCRIPTION
**Pull request by marcin-hoa**
### Changes
- added `proxy.config.json` and included it in hoa-recipes's project.json
- added libs
    - feature-recipe-list
    - data-access-api and needed DTOs
- updated prettier and fixed prettier config

### Questions
- app.component.html now has two imports from libs folder. It will grow over time. Should it also migrate to libs?

### CODE REVIEW

File: libs/data-access-api/src/lib/data-access-api/recipes.repository.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import { Observable } from 'rxjs';
+import { RecipeDto } from './dto/RecipeDto';
+
+@Injectable()
+export class RecipesRepository {
+  private baseUrl: string;
+
+  constructor(private client: HttpClient) {
+    this.baseUrl = `/api/recipes`;
+  }
```
Comment (DDonochVA) [inline libs/data-access-api/src/lib/data-access-api/recipes.repository.ts]: give the new `inject` function a try
```suggestion
  private baseUrl = `/api/recipes`;
  private client = inject(HttpClient);
```

File: libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { CommonModule } from '@angular/common';
+import { Component, OnInit } from '@angular/core';
+import { MatIcon } from '@angular/material/icon';
+import { MatListModule } from '@angular/material/list';
+import { RecipeDto, RecipesRepository } from '@hoa-recipes/data-access-api';
+
+@Component({
```
Comment (DDonochVA) [inline libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts:8]: I would add `OnPush` strategy

File: apps/hoa-recipes/src/app/app.component.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import { Component } from '@angular/core';
 import { MatDialog } from '@angular/material/dialog';
 import { RouterModule } from '@angular/router';
+import { FeatureRecipeListComponent } from '@hoa-recipes/feature-recipe-list';
 import { FeatureTitleBarComponent } from '@hoa-recipes/feature-title-bar';
 
 @Component({
```
Comment (DDonochVA) [inline apps/hoa-recipes/src/app/app.component.ts:6]: I would add `OnPush` strategy

File: libs/feature-title-bar/src/lib/feature-title-bar/feature-title-bar.component.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@ import { MatDialog } from '@angular/material/dialog';
 import { MatIcon } from '@angular/material/icon';
 import { MatToolbarModule } from '@angular/material/toolbar';
 import { UiAuthorDialogComponent } from '@hoa-recipes/ui-author-dialog';
+import { take } from 'rxjs';
 
 @Component({
```
Comment (DDonochVA) [inline libs/feature-title-bar/src/lib/feature-title-bar/feature-title-bar.component.ts]: OnPush

Comment (DDonochVA) [reply libs/feature-title-bar/src/lib/feature-title-bar/feature-title-bar.component.ts]: +make sure if `CommonModule` is necessary

File: libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  selector: 'hoa-recipes-feature-recipe-list',
+  standalone: true,
+  imports: [CommonModule, MatListModule, MatIcon],
```
Comment (DDonochVA) [inline libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts]: make sure if `CommonModule` is necessary

File: apps/hoa-recipes/src/app/app.component.html [hunk 6]
```diff
@@ -1,2 +1,3 @@
 <hoa-recipes-feature-title-bar></hoa-recipes-feature-title-bar>
+<hoa-recipes-feature-recipe-list></hoa-recipes-feature-recipe-list>
```
Comment (DDonochVA) [inline apps/hoa-recipes/src/app/app.component.html]: you can use self-closed tags
```suggestion
<hoa-recipes-feature-recipe-list />
```

File: libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  templateUrl: './feature-recipe-list.component.html',
+  styleUrl: './feature-recipe-list.component.scss',
+})
+export class FeatureRecipeListComponent implements OnInit {
+  protected recipes: RecipeDto[] = [];
+
+  constructor(private recipeRepository: RecipesRepository) {}
+
+  ngOnInit(): void {
+    this.recipeRepository.getAll().subscribe((res) => {
```
Comment (DDonochVA) [inline libs/feature-recipe-list/src/lib/feature-recipe-list/feature-recipe-list.component.ts]: I would recommend managing data with NgRX

**Review by DDonochVA — COMMENTED** [review]
Summary:
- Use `OnPush` strategy in all components
- Injecting can be done via `inject` function instead of constructor
- Avoid importing CommonModule as it was broke up into smaller chunks (NgIf, NgFor)
- Install NgRX

**Comment by marcin-hoa** [comment]
I fixed almost all things. The only one left is NgRx, which I'm gonna try out now. 

I also added `json-server` to replace crudcrud. It works basically the same, but it is unlimited :)

**Pull request merged**